src/lib/ext-activitylog.js

```diff
@@ -3,26 +3,82 @@ import { save } from './save-load.js';
 class Model {
   constructor() {
     this.logs = [];
+    this.filter = {
+      id: [],
+      viewType: [],
+      type: [],
+    };
   }
 
   addNewLogs(logs) {
     this.logs.push(...logs);
   }
+
+  addFilter({ logKey, valueEquals }) {
+    this.filter[logKey].push(valueEquals);
+  }
+
+  removeFilter({ logKey, valueEquals }) {
+    const logKeyIndex = this.filter[logKey].indexOf(valueEquals);
+    if (logKeyIndex > -1) {
+      this.filter[logKey].splice(logKeyIndex, 1);
+    }
+  }
 }
 
 class View {
   constructor() {
-    this.logTable = document.querySelector('log-view');
-    this.saveLogBtn = document.getElementById('saveLogBtn');
+    this.logView = document.querySelector('log-view');
+    this.saveLogBtn = document.querySelector('#saveLogBtn');
     this.notice = document.querySelector('.notice');
 
-    this.saveLogBtn.addEventListener('click', function () {
-      this.dispatchEvent(new CustomEvent('savelog'));
-    });
+    this.extFilter = document.querySelector('filter-option[filter-key="id"]');
+    this.viewTypeFilter = document.querySelector(
+      'filter-option[filter-key="viewType"]'
+    );
+    this.apiTypeFilter = document.querySelector(
+      'filter-option[filter-key="type"]'
+    );
+
+    this.saveLogBtn.addEventListener('click', this);
+  }
+
+  setModelFilter(filterFunc) {
+    this.isModelFilterMatched = filterFunc;
+  }
+
+  handleEvent(event) {
+    if (event.type === 'click') {
+      switch (event.target) {
+        case this.saveLogBtn:
+          this.saveLogBtn.dispatchEvent(new CustomEvent('savelog'));
+          break;
+        default:
+          throw new Error(`wrong event target - ${event.target.tagName}`);
+      }
+    } else {
+      throw new Error(`wrong event type - ${event.type}`);
+    }
   }
 
   addTableRows(logs) {
-    this.logTable.addNewRows(logs);
+    this.logView.addNewRows({
+      logs,
+      isHidden: (log) => this.isModelFilterMatched(log),
+    });
+    this.updateFilterOptions(logs);
+  }
+
+  updateFilterOptions(logs) {
+    this.extFilter.updateFilterCheckboxes(logs);
+    this.viewTypeFilter.updateFilterCheckboxes(logs);
+    this.apiTypeFilter.updateFilterCheckboxes(logs);
+  }
+
+  filterTableRows() {
+    this.logView.filterLogViewItems({
```

Comment:
When a function name or variable name is meaningful in different classes, try to consistently use the same naming.

```javascript
updateLogFilter(filterFunc) {
  this.logView.updateLogFilter(filterFunc);
}
```

(or `setLogFilter`)